### PR TITLE
feat(admin-module): admin module enhancements

### PR DESCRIPTION
**Pull request by brannon-darby**
# FireLoop

### What type of pull request are you creating?
- [ ] Bug Fix
- [x] Enhancement
- [ ] Documentation

### Detailed description of changes:
Updated Admin Module with role mapping and ACLs
### How many unit test did you write for this pull request?
travis

### CODE REVIEW

**Comment by brannon-darby** [comment]
final updates have been made, all checks have passed, and this branch is now live at starter.fireloop.io.  please review and approve.

@jonathan-casarrubias 
@kattsushi

File: api/common/models/fire-user.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "FireUser",
```
Comment (jonathan-casarrubias) [inline api/common/models/fire-user.json:2]: I'm not sure about defining the model as FireUser, I would rather use Account instead, as a starter we should not decide the name of other projects model names

File: api/server/model-config.json [hunk 2]
```diff
@@ -46,5 +54,9 @@
   "Container": {
     "dataSource": "storage",
     "public": true
+  },
+  "FireUser": {
```
Comment (jonathan-casarrubias) [inline api/server/model-config.json:58]: Again I believe Account would be better than FireUser

File: webapp/src/app/admin/admin.component.ts [hunk 3]
```diff
@@ -1,39 +1,60 @@
...
+  constructor(
+    private uiService: UIService,
+    private userApi: FireUserApi
+  ) {
+    this.setSidebarNav();
     this.uiService.setSidebarNav(this.sidebarNav);
   }
+
+  setSidebarNav() {
+    this.sidebarNav = [
```
Comment (jonathan-casarrubias) [inline webapp/src/app/admin/admin.component.ts:24]: Recomiendo usar interface en vez de implicito any.

````js
interface NavItem{
  name: string,
  link: string,
  icon: string
}

private sidebarNav: NavItem[];

this.sidebarNav = [
   ....
]
````

File: webapp/src/app/admin/admin.component.ts [hunk 4]
```diff
@@ -1,39 +1,60 @@
...
-      'icon': 'users'
-    },
-    {
-      'name': 'Roles',
-      'link': '/admin/roles',
-      'icon': 'tags'
-    },
-  ];
+  private authIcon;
+  private sidebarNav: any;
```
Comment (jonathan-casarrubias) [inline webapp/src/app/admin/admin.component.ts:13]: Recomiendo usar interface en vez de implicito any.

````js
interface NavItem{
  name: string,
  link: string,
  icon: string
}

private sidebarNav: NavItem[];

this.sidebarNav = [
   ....
]
````

File: webapp/src/app/admin/controls/control.service.ts [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+import { Subscription } from 'rxjs/Subscription';
+import { FormService } from '../../ui/form/ui-form.service';
+import { Observable } from 'rxjs/Observable';
+import { sortBy } from 'lodash';
+
+@Injectable()
+export class ControlService {
+
+  private subscriptions: Subscription[] = new Array<Subscription>();
+  models: any;
```
Comment (jonathan-casarrubias) [inline webapp/src/app/admin/controls/control.service.ts:12]: Missing interface..

File: webapp/src/app/admin/controls/control.service.ts [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+  roles: any;
```
Comment (jonathan-casarrubias) [inline webapp/src/app/admin/controls/control.service.ts:13]: Missing interface..

File: webapp/src/app/admin/controls/control.service.ts [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    private formService: FormService,
+    private aclApi: ACLApi,
+    private sdkModels: SDKModels,
+    private roleApi: RoleApi
+  ) {
+    this.getModels();
+    this.getRoles();
+  }
+
+  getCardButtons() {
```
Comment (jonathan-casarrubias) [inline webapp/src/app/admin/controls/control.service.ts:25]: Return type is missing

File: webapp/src/app/admin/controls/control.service.ts [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+    return {
+      class: 'btn btn-secondary btn-block float-right',
+      icon: 'plus',
+      text: 'Create'
+    };
+  }
+
+  getTableHeaders() {
```
Comment (jonathan-casarrubias) [inline webapp/src/app/admin/controls/control.service.ts:33]: Return type is missing

File: webapp/src/app/admin/controls/control.service.ts [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+      'Property',
+      'Access Type',
+      'Permission',
+      'Principal Type',
+      'Principal ID',
+      'Actions'
+    ];
+  }
+
+  getFormConfig(formType: string) {
```
Comment (jonathan-casarrubias) [inline webapp/src/app/admin/controls/control.service.ts:45]: Return type is missing

File: webapp/src/app/admin/controls/control.service.ts [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+  }
+
+  getRoles() {
+    this.subscriptions.push(this.roleApi.find().subscribe((roles: any) => (this.roles = roles)));
+  }
+
+  getFormFields(formType: string) {
+    let models = [];
+    let roles = [];
+    const defaultRoles = ['$authenticated', '$everyone', '$owner', '$unauthenticated'];
```
Comment (jonathan-casarrubias) [inline webapp/src/app/admin/controls/control.service.ts:65]: should not this be defined as a class parameter? so it is declared just once... Here every time you call the getFormFields method you redeclare the const defaultRoles which is wrong, a const should not be re-declared

File: webapp/src/app/admin/dashboard/dashboard.component.ts [hunk 11]
```diff
@@ -27,10 +29,12 @@ import { Subscription } from 'rxjs/Subscription';
 
 export class DashboardComponent implements OnDestroy {
   public dashCards: any = [];
```
Comment (jonathan-casarrubias) [inline webapp/src/app/admin/dashboard/dashboard.component.ts:31]: interface?

File: webapp/src/app/ui/ui.service.ts [hunk 12]
```diff
@@ -46,6 +46,10 @@ export class UIService {
     this.sidebarOpen = !this.sidebarOpen;
   }
 
+  getSidebarNav() {
```
Comment (jonathan-casarrubias) [inline webapp/src/app/ui/ui.service.ts:49]: return interface is missing

File: webapp/src/app/ui/ui.service.ts [hunk 13]
```diff
@@ -46,6 +46,10 @@ export class UIService {
...
+    return this.sidebarNav;
+  }
+
   setSidebarNav(nav: any): void {
```
Comment (jonathan-casarrubias) [inline webapp/src/app/ui/ui.service.ts:53]: Nav interface is missing

**Review by jonathan-casarrubias — CHANGES_REQUESTED** [review]

**Pull request closed without merging**